codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -290,11 +266,13 @@ private int maxUsableChannelBytes() {
 
     /**
      * Package private for testing purposes only!
+     *
      * @param requestedBytes The desired amount of bytes.
-     * @return The amount of bytes that can be supported by underlying {@link Channel} without queuing "too-much".
+     * @return The amount of bytes that can be supported by underlying {@link
+     * io.netty.channel.Channel} without queuing "too-much".
      */
-    final int writableBytes(int requestedBytes) {
-        return Math.min(requestedBytes, maxUsableChannelBytes());
+    private int writableBytes() {
```

Comment:
nit: consider keeping `int requestedBytes` parameter so we don't have to continuously re-acquire the state object for the connection stream in the loop. Shouldn't be looping too much...but may save at least 1 lookup in the majority of cases (when we have unwritten bytes).